controllers/airline_controller_test.go

```diff
@@ -35,9 +38,116 @@ func TestHandleAirlineByIdController(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockService := mocks.NewMockIAirlineRepository(mockCtrl)
+	controllerRepo := NewAirlineControllerRepository(mockService)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	airline := models.Airline{
+		Name: "XYZAirline",
+	}
+	mockService.EXPECT().CreateNewAirline(&airline).Return(nil)
+	reqBody := `{"name":"XYZAirline"}`
+
+	ctx.Request, _ = http.NewRequest("POST", "/airline", strings.NewReader(reqBody))
+	controllerRepo.HandleCreateNewAirline(ctx)
+	assert.Equal(t, http.StatusCreated, ctx.Writer.Status())
+
+	var response models.Airline
+	err := json.Unmarshal([]byte(reqBody), &response)
+	assert.NoError(t, err)
+
+	assert.Equal(t, airline.Name, response.Name)
+
+}
+
+func TestHandleCreateNewAirlineWhenTheMandatoryValueIsAbsent(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockService := mocks.NewMockIAirlineRepository(mockCtrl)
+	controllerRepo := NewAirlineControllerRepository(mockService)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	reqBody := `{"Name":""}`
+
+	ctx.Request, _ = http.NewRequest("POST", "/airline", strings.NewReader(reqBody))
+	controllerRepo.HandleCreateNewAirline(ctx)
+	assert.Equal(t, http.StatusBadRequest, ctx.Writer.Status())
+
+}
+
+func TestHandleCreateNewAirlineWhenThePayloadIsEmpty(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockService := mocks.NewMockIAirlineRepository(mockCtrl)
+	controllerRepo := NewAirlineControllerRepository(mockService)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	reqBody := `{}`
+
+	ctx.Request, _ = http.NewRequest("POST", "/airline", strings.NewReader(reqBody))
+	controllerRepo.HandleCreateNewAirline(ctx)
+	assert.Equal(t, http.StatusBadRequest, ctx.Writer.Status())
+
+}
+
+func TestHandleCreateNewAirlineWhenTheMandatoryKeyIsAbsent(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockService := mocks.NewMockIAirlineRepository(mockCtrl)
+	controllerRepo := NewAirlineControllerRepository(mockService)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	reqBody := `{"Count":2}`
+	ctx.Request, _ = http.NewRequest("POST", "/airline", strings.NewReader(reqBody))
+	controllerRepo.HandleCreateNewAirline(ctx)
+	assert.Equal(t, http.StatusBadRequest, ctx.Writer.Status())
+
+}
+
+func TestHandleCreateNewAirlineWhenDataOfDifferentDatatypeIsGiven(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockService := mocks.NewMockIAirlineRepository(mockCtrl)
+	controllerRepo := NewAirlineControllerRepository(mockService)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	reqBody := `{"name":123}`
+
+	ctx.Request, _ = http.NewRequest("POST", "/airline", strings.NewReader(reqBody))
+	controllerRepo.HandleCreateNewAirline(ctx)
```

Comment:
need an empty line to differentiate between "assemble", "execute" and "assert" group of actions

Reply:
okay